### PR TITLE
[actions] adding github actions for tests with changes for pytest

### PR DESCRIPTION
**Pull request by Harveyt47**
Hi @mmcky this has the GitHub action for tests with `pytest`. It also has the required changes to the test file for `pytest` to run. 
PS sorry for al the commits didn't realise it tracked them from another branch I accidently merged to my forks master.

### CODE REVIEW

File: .github/workflow/test.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: continuous-integration
+
+on:
+  push:
+    branches:
+      - master
+
+  pull_request:
+    branches:
+      - master
```
Comment (mmcky) [inline .github/workflow/test.yml:10]: @Harveyt47 this should be activated on pull-requests in general and not restricted to `master`

File: .github/workflow/test.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+jobs:
+  tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest]
```
Comment (mmcky) [inline .github/workflow/test.yml:16]: I think we can test on `os x ` as well

File: .github/workflow/test.yml (hunk 1 above)
Comment (mmcky) [reply .github/workflow/test.yml:10]: `Github Actions` should show up once you make this change

**Comment by mmcky** [comment]
- [ ] will need to ensure we `Squash and Merge` this when it is ready.

**Comment by mmcky** [comment]
@Harveyt47 

We should also tidy up the `travis` files as part of this PR as `travis` will no longer need to be run once `Github Actions` takes over.

**Comment by Harveyt47** [comment]
> @Harveyt47
> We should also tidy up the travis files as part of this PR as travis will no longer need to be run once Github Actions takes over.

Is it just`.travis.yml` and `.coveralls.yml` that need to be removed?

File: .github/workflow/test.yml (hunk 1 above)
Comment (Harveyt47) [reply .github/workflow/test.yml]: Hi @mmcky I've done this and I don't see `GitHub Actions` showing up

**Comment by mmcky** [comment]
> > We should also tidy up the travis files as part of this PR as travis will no longer need to be run once Github Actions takes over.
> 
> @mmcky is it just`.travis.yml` and `.coveralls.yml` that need to be removed?

hey @Harveyt47 lets not remove `.coveralls.yml`. Will we get `coveralls` results from `Github Actions` or was this linked to `travis`?

File: .github/workflow/test.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+name: continuous-integration
+
+on: [pull_request, push]
```
Comment (mmcky) [inline .github/workflow/test.yml:3]: perhaps this should be 

```yml
on: [pull_request | push]
```

**Comment by Harveyt47** [comment]
> We should also tidy up the travis files as part of this PR as travis will no longer need to be run once Github Actions takes over.
> 
> @mmcky is it just.travis.yml and .coveralls.yml that need to be removed?
> 
> hey @Harveyt47 lets not remove .coveralls.yml. Will we get coveralls results from Github Actions or was this linked to travis?

@mmcky I think `.coveralls.yml` was linked to `travis`. It only contains a `repo token` and a `service-name` which is `travis`

**Comment by mmcky** [comment]
Also here is some info for `coveralls` on `github actions` 

https://github.com/marketplace/actions/coveralls-github-action

**Comment by mmcky** [comment]
hey @Harveyt47 -- I think the path is incorrect for `.github/workflow` should be `.github/workflows` with an `s`

**Comment by Harveyt47** [comment]
> hey @Harveyt47 -- I think the path is incorrect for .github/workflow should be .github/workflows with an s

Ahh yes, sorry

**Comment by mmcky** [comment]
@Harveyt47 I am closing this PR given the issues with `Github Actions` triggering from a Fork. Please re-submit via a branch. Thanks.

**Pull request closed without merging**